packages/tkg-clusterclass-aws/bundle/config/upstream/overlay-pod-security-admission.yaml

```
@@ -64,71 +64,104 @@ spec:
-                        {{ .podSecurityStandard.audit }}
-                      {{- else -}}
-                        {{ $defaultWarnAudit }}
-                      {{- end }}"
-                    audit-version: "{{ .podSecurityStandard.auditVersion -}}"
-                    warn: "{{ if .podSecurityStandard.warn -}}
-                        {{ .podSecurityStandard.warn }}
-                      {{- else -}}
-                        {{ $defaultWarnAudit }}
-                      {{- end }}"
-                    warn-version: "{{ .podSecurityStandard.warnVersion -}}"
-                  exemptions:
-                    usernames: []
-                    runtimeClasses: []
-                    namespaces: [{{ $namespace_exemptions }}]
-            path: /etc/kubernetes/kube-apiserver-admission-pss.yaml
-    enabledIf: '{{ and (not .podSecurityStandard.deactivated) (semverCompare ">= v1.24" .builtin.controlPlane.version) }}'
+  - name: apiserverEventRateLimitConf
+    required: false
+    schema:
+      openAPIV3Schema:
+        type: string
+        default: ""
+  #@overlay/match missing_ok=True
+  patches:
+    #@overlay/remove
+    #@overlay/match by=overlay.map_key("name"),missing_ok=True
+    - name: admissionConfiguration
+    #@overlay/append
+    - name: admissionConfiguration
+      description: "Adds an admission configuration for kube-apiserver."
+      definitions:
+        - selector:
+            apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+            kind: KubeadmControlPlaneTemplate
+            matchResources:
+              controlPlane: true
+          jsonPatches:
+            - op: add
+              path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraArgs/admission-control-config-file"
+              valueFrom:
+                template: |
+                  {{ $enableAdmissionPlugin := false }}
+                  {{- range $key, $val := .apiServerExtraArgs }}
+                  {{- if eq $key "enable-admission-plugin" }}
+                  {{- $enableAdmissionPlugin = true }}
+                  {{- end }}
+                  {{- if not $enableAdmissionPlugin }}
+                  enable-admission-plugins: "NodeRestriction,EventRateLimit"
+                  {{- end }}
+                  admission-control-config-file: "/etc/kubernetes/admission-control-config.yaml"
+            - op: add
+              path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes/-"
+              valueFrom:
+                template: |
+                  - name: admin-control-conf
+                    hostPath: /etc/kubernetes/admission-control-config.yaml
+                    mountPath: /etc/kubernetes/admission-control-config.yaml
+                  {{ if .apiserverEventRateLimitConf }}
+                  - name: event-conf
+                    hostPath: /etc/kubernetes/eventconfig.yaml
+                    mountPath: /etc/kubernetes/eventconfig.yaml
+                  {{- end}}
```

Also needs to get separated and event-conf get added in separate patch, where `enabled_if` decides if it is to add or not
```suggestion
            - op: add
              path: "/spec/template/spec/kubeadmConfigSpec/clusterConfiguration/apiServer/extraVolumes/-"
              valueFrom:
                template: |
                  name: admin-control-conf
                  hostPath: /etc/kubernetes/admission-control-config.yaml
                  mountPath: /etc/kubernetes/admission-control-config.yaml
```